### PR TITLE
feat: add acceptExchangeRateUpdateHook and return gas info during route execution

### PR DESCRIPTION
**Pull request by chybisov**

### CODE REVIEW

File: src/utils/parseError.ts [hunk 1]
```diff
@@ -95,6 +95,14 @@ export const getTransactionFailedMessage = (
     : baseString
 }
 
+export const getExchangeRateChangedMessage = (step: Step) => {
```
Comment (Addminus) [inline src/utils/parseError.ts:98]: With the new hook, this error message feels off. Especially with the premise, that the exchangerate hook is not really covering excessive slippage anymore. it just uses it as a threshold.

Maybe something like "Transaction was not sent, your funds are still in your wallet. The exchange rate has changed and the previous estimation can not be fulfilled due to severe value loss."

Comment (Addminus) [reply src/utils/parseError.ts:98]: This would cover cases when no hook is set, but also explain why the transaction failed when the user declined

**Review by Addminus — CHANGES_REQUESTED** [review]
Looks good. I have a question regarding the exchange rate error message

File: src/utils/parseError.ts (hunk 1 above)
Comment (chybisov) [reply src/utils/parseError.ts]: @Addminus done. I removed 'severe' though because it can be only 0.1% sometimes 😄

**Pull request merged**